# Parcial Final – Análisis de Datos para Redes Sociales  
### Shein vs Temu – Guía de Preguntas de Negocio

Este cuaderno está diseñado para acompañar el **parcial final**.  
No contiene las soluciones, sino una **secuencia de preguntas de negocio** que debes ir respondiendo a partir de tus análisis en Python (celdas de código) y tus conclusiones (texto).

> **Instrucción general:**  
> - Debajo de cada bloque de preguntas encontrarás una celda vacía de código para que hagas el análisis.  
> - Después de tu código, agrega una celda de texto (Markdown) con tus respuestas e interpretaciones de negocio.


In [ ]:
!wget https://github.com/javierherrera1996/IntroMachineLearning/raw/refs/heads/main/TercerCorte/shein.csv

--2025-11-26 23:46:42--  https://github.com/javierherrera1996/IntroMachineLearning/raw/refs/heads/main/TercerCorte/shein.csv
Resolving github.com (github.com)... 140.82.114.3
Connecting to github.com (github.com)|140.82.114.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/javierherrera1996/IntroMachineLearning/refs/heads/main/TercerCorte/shein.csv [following]
--2025-11-26 23:46:42--  https://raw.githubusercontent.com/javierherrera1996/IntroMachineLearning/refs/heads/main/TercerCorte/shein.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1760315 (1.7M) [text/plain]
Saving to: ‘shein.csv’

shein.csv           100%[===================>]   1.68M  --.-KB/s    in 0.06s   

2025-11-26 23:46:42 (28

In [ ]:
!wget https://github.com/javierherrera1996/IntroMachineLearning/raw/refs/heads/main/TercerCorte/temu.csv

--2025-11-26 23:46:42--  https://github.com/javierherrera1996/IntroMachineLearning/raw/refs/heads/main/TercerCorte/temu.csv
Resolving github.com (github.com)... 140.82.112.4
Connecting to github.com (github.com)|140.82.112.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/javierherrera1996/IntroMachineLearning/refs/heads/main/TercerCorte/temu.csv [following]
--2025-11-26 23:46:42--  https://raw.githubusercontent.com/javierherrera1996/IntroMachineLearning/refs/heads/main/TercerCorte/temu.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 346171 (338K) [text/plain]
Saving to: ‘temu.csv’

temu.csv            100%[===================>] 338.06K  --.-KB/s    in 0.03s   

2025-11-26 23:46:42 (10.2 MB

In [ ]:
import pandas as pd


In [ ]:
!pip install -qU requests pandas google-generativeai nltk seaborn wordcloud scikit-learn

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 91.2/91.2 kB 2.1 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 64.7/64.7 kB 3.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.4/12.4 MB 76.9 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.5/1.5 MB 47.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 9.5/9.5 MB 79.9 MB/s eta 0:00:00
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
google-colab 1.0.0 requires pandas==2.2.2, but you have pandas 2.3.3 which is incompatible.
google-colab 1.0.0 requires requests==2.32.4, but you have requests 2.32.5 which is incompatible.


In [ ]:
shein = pd.read_csv('shein.csv')
shein.head()

alt audioUrl  \
0  Photo by shein_co on November 25, 2025. May be...      NaN   
1  Photo by shein_co on September 26, 2025. May b...      NaN   
2  Photo by shein_co on October 03, 2025. May be ...      NaN   
3  Photo by shein_co on November 26, 2025. May be...      NaN   
4  Photo by shein_co on September 19, 2025. May b...      NaN   

                                             caption  \
0  ¿De compras para encontrar las mejores ofertas...   
1  Recién recogido y fabuloso—¿qué vestido es tu ...   
2  Una energía de un solo tono, esa flexión codif...   
3  Todo lo que escuchamos fue "quédate en casa" y...   
4  La única necesidad que tengo para poder termin...   

                                    childPosts/0/alt  childPosts/0/caption  \
0  Photo by shein_co on November 25, 2025. May be...                   NaN   
1  Photo by shein_co on September 26, 2025. May b...                   NaN   
2  Photo by shein_co on October 03, 2025. May be ...                   NaN   
3                                                NaN                   NaN   
4                                                NaN                   NaN   

   childPosts/0/commentsCount  childPosts/0/dimensionsHeight  \
0                         0.0                         1350.0   
1                         0.0                         1350.0   
2                         0.0                         1350.0   
3                         NaN                            NaN   
4                         NaN                            NaN   

   childPosts/0/dimensionsWidth  \
0                        1080.0   
1                        1080.0   
2                        1080.0   
3                           NaN   
4                           NaN   

                             childPosts/0/displayUrl  \
0  https://scontent-atl3-2.cdninstagram.com/v/t51...   
1  https://scontent-hou1-1.cdninstagram.com/v/t51...   
2  https://scontent-ord5-2.cdninstagram.com/v/t51...   
3                                                NaN   
4                                                NaN   

   childPosts/0/firstComment  ...   ownerUsername  productType    shortCode  \
0                        NaN  ...  shein_colombia          NaN  DRetnrflmeB   
1                        NaN  ...  shein_colombia          NaN  DPEXbp5jtSe   
2                        NaN  ...  shein_colombia          NaN  DPWh7lTASw0   
3                        NaN  ...  shein_colombia          NaN  DRhbjBhgaUs   
4                        NaN  ...  shein_colombia          NaN  DOycs74DXgF   

                  timestamp     type  \
0  2025-11-25T12:46:02.000Z  Sidecar   
1  2025-09-26T14:08:01.000Z  Sidecar   
2  2025-10-03T15:26:01.000Z  Sidecar   
3  2025-11-26T14:06:02.000Z    Image   
4  2025-09-19T15:08:01.000Z    Image   

                                        url videoDuration videoPlayCount  \
0  https://www.instagram.com/p/DRetnrflmeB/           NaN            NaN   
1  https://www.instagram.com/p/DPEXbp5jtSe/           NaN            NaN   
2  https://www.instagram.com/p/DPWh7lTASw0/           NaN            NaN   
3  https://www.instagram.com/p/DRhbjBhgaUs/           NaN            NaN   
4  https://www.instagram.com/p/DOycs74DXgF/           NaN            NaN   

   videoUrl  videoViewCount  
0       NaN             NaN  
1       NaN             NaN  
2       NaN             NaN  
3       NaN             NaN  
4       NaN             NaN  

[5 rows x 633 columns]

In [ ]:
temu = pd.read_csv('temu.csv')
temu.head()

alt  \
0  Photo by TemuColombia on July 11, 2025.   
1                                      NaN   
2                                      NaN   
3                                      NaN   
4  Photo by TemuColombia on July 11, 2025.   

                                            audioUrl  \
0                                                NaN   
1  https://scontent-iad3-1.cdninstagram.com/o1/v/...   
2  https://scontent-lga3-1.cdninstagram.com/o1/v/...   
3  https://scontent-lga3-2.cdninstagram.com/o1/v/...   
4                                                NaN   

                                             caption  commentsCount  \
0  ¡Alerta de promoción! Descubre productos que s...             15   
1  ❤️❤️Valen mucho la pena⏰  #temu #compraentemu ...              1   
2  ❤️❤️Valen mucho la pena⏰  #temu #compraentemu ...              7   
3  ¡Los aperitivos reinventados! Nuestro anillo d...            359   
4  Alerta de Rebajas💥 ¡No te pierdas el Gran Even...             75   

   dimensionsHeight  dimensionsWidth  \
0              1080             1080   
1              1920             1080   
2              1920             1080   
3              1333              750   
4              1080             1080   

                                          displayUrl  \
0  https://scontent-lax7-1.cdninstagram.com/v/t51...   
1  https://scontent-iad3-1.cdninstagram.com/v/t51...   
2  https://scontent-lga3-2.cdninstagram.com/v/t51...   
3  https://scontent-lga3-3.cdninstagram.com/v/t51...   
4  https://scontent-dfw5-1.cdninstagram.com/v/t51...   

                                        firstComment hashtags/0  \
0  🤬🤬🤬TENGO UN PAQUETE PARA DEVOLUCIÓN ESPERANDO ...        NaN   
1                                            Lindas!       temu   
2  Que  tamaño tienen los papa noeles que suben y...       temu   
3  🤬🤬🤬TENGO UN PAQUETE PARA DEVOLUCIÓN ESPERANDO ...       temu   
4                    100% falso este servicio pésimo       temu   

      hashtags/1  ...  ownerUsername productType    shortCode  \
0            NaN  ...  temucolombia_         NaN  DL9iB1-tcFn   
1   compraentemu  ...  temucolombia_       clips  DRUUa95DAvf   
2   compraentemu  ...  temucolombia_       clips  DRUUc2Pj6a8   
3            NaN  ...  temucolombia_       clips  C4uC0VShghw   
4  hallazgostemu  ...  temucolombia_         NaN  DL9iB55Nskv   

                  timestamp   type                                       url  \
0  2025-07-11T08:51:09.000Z  Image  https://www.instagram.com/p/DL9iB1-tcFn/   
1  2025-11-21T11:52:32.000Z  Video  https://www.instagram.com/p/DRUUa95DAvf/   
2  2025-11-21T11:52:27.000Z  Video  https://www.instagram.com/p/DRUUc2Pj6a8/   
3  2024-03-20T02:33:20.000Z  Video  https://www.instagram.com/p/C4uC0VShghw/   
4  2025-07-11T08:51:10.000Z  Image  https://www.instagram.com/p/DL9iB55Nskv/   

   videoDuration  videoPlayCount  \
0            NaN             NaN   
1         16.939         26146.0   
2         13.845         20891.0   
3          8.933        615177.0   
4            NaN             NaN   

                                            videoUrl  videoViewCount  
0                                                NaN             NaN  
1  https://scontent-iad3-1.cdninstagram.com/o1/v/...          7599.0  
2  https://scontent-lga3-3.cdninstagram.com/o1/v/...          5273.0  
3  https://scontent-lga3-2.cdninstagram.com/o1/v/...        262119.0  
4                                                NaN             NaN  

[5 rows x 257 columns]

In [ ]:
shein

alt  \
0    Photo by shein_co on November 25, 2025. May be...   
1    Photo by shein_co on September 26, 2025. May b...   
2    Photo by shein_co on October 03, 2025. May be ...   
3    Photo by shein_co on November 26, 2025. May be...   
4    Photo by shein_co on September 19, 2025. May b...   
..                                                 ...   
195            Photo by shein_co on November 18, 2024.   
196                 Photo by shein_co on May 10, 2025.   
197                                                NaN   
198                                                NaN   
199                                                NaN   

                                              audioUrl  \
0                                                  NaN   
1                                                  NaN   
2                                                  NaN   
3                                                  NaN   
4                                                  NaN   
..                                                 ...   
195                                                NaN   
196                                                NaN   
197  https://scontent-atl3-1.cdninstagram.com/o1/v/...   
198  https://scontent-bos5-1.cdninstagram.com/o1/v/...   
199  https://instagram.fmem1-2.fna.fbcdn.net/o1/v/t...   

                                               caption  \
0    ¿De compras para encontrar las mejores ofertas...   
1    Recién recogido y fabuloso—¿qué vestido es tu ...   
2    Una energía de un solo tono, esa flexión codif...   
3    Todo lo que escuchamos fue "quédate en casa" y...   
4    La única necesidad que tengo para poder termin...   
..                                                 ...   
195  Los días más bonitos vienen con suéteres lindo...   
196  Hay abrazos que salvan, miradas que consuelan ...   
197  Se acabaron los pretextos para saltarnos las v...   
198  La funda que elijas dice mucho de ti... ¿mi co...   
199  Las tiendas que @miscanola ha descubierto son ...   

                                      childPosts/0/alt  childPosts/0/caption  \
0    Photo by shein_co on November 25, 2025. May be...                   NaN   
1    Photo by shein_co on September 26, 2025. May b...                   NaN   
2    Photo by shein_co on October 03, 2025. May be ...                   NaN   
3                                                  NaN                   NaN   
4                                                  NaN                   NaN   
..                                                 ...                   ...   
195                                                NaN                   NaN   
196                                                NaN                   NaN   
197                                                NaN                   NaN   
198                                                NaN                   NaN   
199                                                NaN                   NaN   

     childPosts/0/commentsCount  childPosts/0/dimensionsHeight  \
0                           0.0                         1350.0   
1                           0.0                         1350.0   
2                           0.0                         1350.0   
3                           NaN                            NaN   
4                           NaN                            NaN   
..                          ...                            ...   
195                         NaN                            NaN   
196                         NaN                            NaN   
197                         NaN                            NaN   
198                         NaN                            NaN   
199                         NaN                            NaN   

     childPosts/0/dimensionsWidth  \
0                          1080.0   
1                          1080.0   
2                          1080.0   
3                             NaN   
4                    

In [ ]:
temu

alt  \
0   Photo by TemuColombia on July 11, 2025.   
1                                       NaN   
2                                       NaN   
3                                       NaN   
4   Photo by TemuColombia on July 11, 2025.   
5                                       NaN   
6                                       NaN   
7                                       NaN   
8                                       NaN   
9                                       NaN   
10                                      NaN   
11                                      NaN   
12                                      NaN   
13                                      NaN   
14                                      NaN   
15                                      NaN   
16                                      NaN   
17                                      NaN   
18                                      NaN   
19                                      NaN   
20                                      NaN   
21                                      NaN   
22                                      NaN   
23                                      NaN   
24                                      NaN   
25                                      NaN   
26                                      NaN   
27                                      NaN   

                                             audioUrl  \
0                                                 NaN   
1   https://scontent-iad3-1.cdninstagram.com/o1/v/...   
2   https://scontent-lga3-1.cdninstagram.com/o1/v/...   
3   https://scontent-lga3-2.cdninstagram.com/o1/v/...   
4                                                 NaN   
5   https://scontent-atl3-3.cdninstagram.com/o1/v/...   
6   https://scontent-lax7-1.cdninstagram.com/o1/v/...   
7   https://scontent-hou1-1.cdninstagram.com/o1/v/...   
8   https://scontent-atl3-1.cdninstagram.com/o1/v/...   
9   https://scontent-hou1-1.cdninstagram.com/o1/v/...   
10  https://scontent-bos5-1.cdninstagram.com/o1/v/...   
11  https://scontent-dfw5-1.cdninstagram.com/o1/v/...   
12  https://scontent-iad3-1.cdninstagram.com/o1/v/...   
13  https://scontent-iad3-1.cdninstagram.com/o1/v/...   
14  https://scontent-lga3-2.cdninstagram.com/o1/v/...   
15  https://scontent-mia3-3.cdninstagram.com/o1/v/...   
16  https://scontent-mia5-1.cdninstagram.com/o1/v/...   
17  https://scontent-iad3-1.cdninstagram.com/o1/v/...   
18  https://instagram.fhnl3-1.fna.fbcdn.net/o1/v/t...   
19  https://scontent-sea5-1.cdninstagram.com/o1/v/...   
20  https://scontent-hou1-1.cdninstagram.com/o1/v/...   
21  https://scontent-ord5-3.cdninstagram.com/o1/v/...   
22  https://scontent-iad3-2.cdninstagram.com/o1/v/...   
23  https://scontent-lga3-1.cdninstagram.com/o1/v/...   
24  https://scontent-lga3-1.cdninstagram.com/o1/v/...   
25  https://scontent-mia3-2.cdninstagram.com/o1/v/...   
26  https://scontent-ord5-1.cdninstagram.com/o1/v/...   
27  https://instagram.fcha1-1.fna.fbcdn.net/o1/v/t...   

                                              caption  commentsCount  \
0   ¡Alerta de promoción! Descubre productos que s...             15   
1   ❤️❤️Valen mucho la pena⏰  #temu #compraentemu ...              1   
2   ❤️❤️Valen mucho la pena⏰  #temu #compraentemu ...              7   
3   ¡Los aperitivos reinventados! Nuestro anillo d...            359   
4   Alerta de Rebajas💥 ¡No te pierdas el Gran Even...             75   
5   ❤️❤️Valen mucho la pena⏰ \n\nSolo para los nue...             64   
6   🌌👼🏻✨ ¿Quién dijo que volar es de ensueño?¡Solo...            218   
7   ¿Harto de usar múltiples utensilios de limpiez...            201   
8   ❤️❤️Valen mucho la pena⏰  #temu #compraentemu ...              9   
9   ¡Mantente conectado con estilo y haz una decla...            202   
10  ❤️❤️Valen mucho la pena⏰ \n\nSolo para los nue...             94   
11  ¡Empieza el día con energía! 🌱💪 ¡Recárgate y e...           1500   
12  Fruta y ensalada, todo en uno 🍓🥗 Un plato deli...            133   
13  Organiza t

## 2. Ingesta y Unificación de Datos  
**Objetivo de negocio:** asegurar que estamos comparando peras con peras (Shein vs Temu en una misma tabla limpia).

### 2.1. Preguntas de negocio

1. **Foto grande del dataset**

   - ¿Cuántas publicaciones tenemos de Shein y cuántas de Temu? **R/ Shein: 200, 633.  Temu: 28, 257**
  
   - ¿Esta diferencia de volumen ya nos dice algo de la estrategia de cada marca? **R/ Sí. El volumen mucho mayor de Shein indica una estrategia de presencia constante y saturación, donde la marca apuesta por publicar masivamente para dominar el feed y mantener visibilidad. Temu, con un volumen mucho menor, parece seguir una estrategia más selectiva o táctica, enfocada en menos piezas pero posiblemente de mayor control o foco en performance. Esta diferencia ya sugiere que ambas marcas juegan con lógicas distintas de contenido.**

2. **Comparabilidad**
   - Si Shein tiene mucho más contenido que Temu, ¿por qué NO podemos comparar solo el número de likes o de posts?  
   - ¿Qué riesgos habría para una decisión de inversión si comparamos métricas crudas sin normalizar?

**R/ No podemos comparar likes o número de posts porque Shein publica muchísimo más, y por volumen siempre va a “ganar” aunque su contenido no sea más eficiente. Si comparamos métricas crudas sin normalizar, el riesgo es tomar una mala decisión de inversión, premiando a la marca que más publica en vez de la que convierte mejor cada vista en interacción. Se debe comparar por eficiencia (engagement por post o por vista), no por cantidad.**


3. **Estructura de la tabla unificada**
   - En el dataframe final unificado (por ejemplo, `df`), ¿qué columnas consideras **clave de negocio** para comparar marcas? (ej: `brand`, `likes`, `comments`, `views`, `published_at`, etc.).  **R/ Son claves porque brand permite diferenciar cada empresa, likes, comments y views miden el desempeño real del contenido, published_at permite analizar horarios y patrones de publicación, y caption ayuda a entender el tipo de contenido o mensaje. Con estas columnas podemos comparar impacto, eficiencia y estrategia entre Shein y Temu de forma coherente.**

   - Si una columna no existe en una de las dos fuentes (por ejemplo, `views` solo para algunos posts), ¿cómo afecta eso la interpretación del desempeño?
   **R/ Si una columna no existe o está incompleta en una de las marcas, la comparación queda sesgada, porque una marca tendría métricas completas y la otra no. Eso puede hacer que parezca que una rinde peor o mejor sin que sea cierto. En ese caso, la métrica pierde confiabilidad para comparar desempeño y toca aclararlo o evitar usarla para conclusiones.**


4. **Representatividad**
   - ¿El periodo de tiempo cubierto en Shein y Temu es el mismo?  
   - Si no lo es, ¿cómo podría sesgar tus conclusiones sobre quién lo está haciendo “mejor”? **R/ No, los periodos no son exactamente iguales. Shein tiene datos desde noviembre de 2024 hasta noviembre de 2025, mientras que Temu cubre un rango más amplio desde marzo de 2024. Esta diferencia puede sesgar la comparación, porque cada marca puede estar operando en temporadas distintas, campañas diferentes o momentos de mayor o menor demanda. Por eso, una marca podría parecer que “lo hace mejor” simplemente porque estuvo activa en un periodo más favorable, no necesariamente porque su contenido sea más eficiente.**


In [ ]:
temu.shape

(28, 257)

In [ ]:
shein.shape

(200, 633)

In [ ]:
shein['published_at'] = pd.to_datetime(shein['timestamp'], errors='coerce')
temu['published_at'] = pd.to_datetime(temu['timestamp'], errors='coerce')

print("Shein:", shein['published_at'].min(), "→", shein['published_at'].max())
print("Temu:", temu['published_at'].min(), "→", temu['published_at'].max())


Shein: 2024-11-06 15:09:00+00:00 → 2025-11-26 14:06:02+00:00
Temu: 2024-03-14 15:13:55+00:00 → 2025-11-21 11:52:41+00:00


## 3. Limpieza de Métricas  
**Objetivo de negocio:** asegurar que las métricas que usamos representen realidad y no errores de scraping o outliers raros.

### 3.1. Preguntas de negocio

1. **Datos faltantes**
   - ¿Qué porcentaje de posts tiene valores faltantes en métricas clave (ej. `likes`, `comments`, `views`)?
   - Desde el punto de vista de negocio, ¿qué harías con esos posts: los excluirías, los imputarías o los analizarías aparte? ¿Por qué?

   **R/ En Shein, el 60.5% de los posts no tiene dato de videoViewCount, mientras que en Temu solo faltan views en 7.1% de los casos. Desde negocio, excluiría estos posts en el análisis de eficiencia, porque sin views no se puede calcular engagement rate y cualquier imputación distorsionaría los resultados. Los de Shein los analizaría aparte como imágenes o formatos no comparables, y los pocos casos faltantes de Temu se pueden remover sin afectar el análisis.**

2. **Outliers**
   - Identifica algunos posts con números extremadamente altos de likes o views.  
   - ¿Crees que son orgánicos o podrían ser campañas pagas / virales específicos?  
   - ¿Cómo cambia tu lectura si incluyes vs. excluyes estos outliers?

   **R/ Los posts con números extremadamente altos de views (entre 500 mil y más de 1.4 millones) en Shein y Temu son outliers claros. Es muy probable que correspondan a campañas pagas, colaboraciones con creadores grandes o momentos virales puntuales, más que a comportamiento orgánico normal. Si los incluyo, inflan los promedios y pueden hacer que una marca parezca “mejor” de lo que realmente es; si los excluyo, la lectura refleja mejor el rendimiento típico del día a día. Lo más razonable es analizarlos aparte para no distorsionar la comparación entre marcas.**

3. **Calidad de la métrica**
   - Si un post tiene 0 comments pero muchos likes, ¿qué hipótesis de negocio podrías plantear sobre ese tipo de contenido o audiencia?  
   - ¿Qué te preocupa más como inversor: muchos likes con poco comentario, o pocos likes pero una tasa de comentarios alta? Explica.

   **R/ Si un post tiene 0 comments pero muchos likes, podría indicar que es contenido **muy visual, rápido de consumir** y que no invita a conversación; la audiencia lo ve, lo aprueba, pero no se detiene a interactuar. Como inversor, me preocupa más tener **muchos likes y pocos comentarios**, porque eso sugiere **interacción superficial**. En cambio, aunque haya menos likes, una tasa de comentarios alta indica **mayor compromiso real**, conversación y una comunidad más conectada con la marca.**



In [ ]:
metrics = ['likesCount', 'commentsCount', 'videoViewCount']


shein[metrics].describe()



likesCount  commentsCount  videoViewCount
count    200.000000     200.000000    7.900000e+01
mean    1989.730000      22.590000    5.189595e+04
std     6327.524509     162.843273    2.094628e+05
min       23.000000       0.000000    2.860000e+02
25%      121.000000       2.000000    1.958000e+03
50%      303.500000       5.000000    3.868000e+03
75%     1332.750000      11.250000    2.946500e+04
max    69638.000000    2270.000000    1.767106e+06

In [ ]:
  metrics = ['likesCount', 'commentsCount', 'videoViewCount']


  temu[metrics].describe()


likesCount  commentsCount  videoViewCount
count    28.000000      28.000000        26.00000
mean   2649.035714     148.321429    146201.50000
std    1715.625628     276.269602     77151.42796
min      95.000000       1.000000      5273.00000
25%    1256.000000      56.250000    120228.50000
50%    2648.000000      81.000000    158033.50000
75%    3616.750000     141.000000    186194.00000
max    6496.000000    1500.000000    262119.00000

## 4. Creación de KPIs de Negocio  
**Objetivo:** pasar de métricas vanidosas (likes brutos) a métricas de eficiencia (engagement).

Supongamos que creas un indicador como:

\>
\> `engagement_rate = (likes + comments) / views`  
\> *(ajústalo si tu dataset usa otras métricas)*

### 4.1. Preguntas de negocio

1. **Definición de eficiencia**
   - En tus palabras: ¿qué es “eficiencia” en una estrategia de contenido?
   **R/ “Eficiencia” es qué tan bien un contenido convierte su alcance en interacción real. No importa solo cuánta gente ve un post, sino cuánta gente realmente reacciona (like o comenta).**

   - ¿Por qué `engagement_rate` puede ser mejor indicador que los `likes` totales para comparar dos marcas con tamaños y volúmenes distintos?
   
   **R/ Porque el engagement_rate ajusta por vistas, mientras que los likes totales dependen del tamaño y del volumen de cada marca. Una marca que publica más o tiene más alcance siempre tendrá más likes brutos, pero eso no quiere decir que su contenido sea mejor. El engagement_rate permite comparar eficiencia, no cantidad.**


2. **Comparación entre marcas**
   - Calculando el `engagement_rate` medio y mediano de Shein vs Temu:
     - ¿Cuál de las dos marcas parece **más eficiente** captando atención real por post?  
     **R/ Con un engagement_rate promedio cercano a 0.37 en Shein y 0.036 en Temu, Shein es claramente más eficiente captando atención real por post. Cada vista genera mucha más interacción en Shein que en Temu.**

     - Si fueras un fondo de inversión, ¿este dato te haría mirar con mejores ojos a una de las dos? ¿Por qué?

     **R/ Como fondo de inversión, este dato me haría mirar con mejores ojos a Shein, porque no solo publica más, sino que su contenido funciona mejor por unidad de alcance. Es decir, convierte el tráfico en interacción real de forma mucho más eficiente.**

3. **Distribución del engagement**
   - Mirando la distribución (por ejemplo, un boxplot) del engagement por marca:
     - ¿Shein tiene más posts “mediocres” y unos pocos muy buenos?  
     - ¿Temu tiene menos posts pero más consistentes?  
     - ¿Qué tipo de distribución preferirías si fueras CMO?

     **R/ La distribución sugiere que Shein tiene muchos posts con engagement bajo o “mediocre” y unos pocos que sobresalen muchísimo (sus outliers), lo que indica un rendimiento más disperso. En cambio, Temu tiene menos posts pero una distribución más estable y consistente, sin picos tan extremos.**

     **Si fuera CMO, preferiría una distribución consistente, porque es más predecible y permite planear mejor. Sin embargo, también valoraría tener algunos “hits” virales como los de Shein si el objetivo es maximizar alcance puntual.**

4. **Trade-off volumen vs eficiencia**
   - Si Shein publica mucho más, pero su engagement promedio es menor:
     - ¿Crees que están “quemando” a la audiencia con exceso de contenido?  
     - ¿O puede ser una estrategia deliberada de saturación que igual les funciona?

     **R/ Si Shein publica mucho más pero su engagement promedio es menor, pueden estar quemando a la audiencia con exceso de contenido, lo que baja la interacción por post. Pero también podría ser una estrategia deliberada de saturación, donde la marca prioriza presencia constante más que eficiencia puntual. Ambas lecturas son plausibles: la primera explica el desgaste en engagement, y la segunda muestra que aun con posts menos eficientes, el volumen total puede seguirles funcionando para mantener visibilidad masiva.**

In [ ]:
shein['engagement_rate'] = (shein['likesCount'] + shein['commentsCount']) / shein['videoViewCount']
shein['engagement_rate'].mean()

np.float64(0.3715771799718908)

In [ ]:
temu['engagement_rate'] = (shein['likesCount'] + shein['commentsCount']) / shein['videoViewCount']
temu['engagement_rate'].mean()

np.float64(0.03683503661103258)

## 5. Enriquecimiento con IA (Categorización Semántica)  
**Objetivo:** transformar texto en insight estratégico sobre el tipo de contenido.

Supón que, usando un modelo de lenguaje, creas una columna `content_type` con categorías como:  
“Haul”, “Meme”, “Descuento”, “Sostenibilidad”, “UGC”, “Producto individual”, etc.

### 5.1. Preguntas de negocio

1. **Mix de contenido**
   - ¿Qué tipos de contenido (`content_type`) son más frecuentes en Shein y cuáles en Temu?  
   - ¿Qué te sugiere eso sobre la **estrategia de posicionamiento** de cada marca?

2. **Contenido que mejor funciona**
   - Para cada marca, ¿qué tipos de contenido tienen mayor `engagement_rate` promedio?  
   - ¿La marca está apostando más por el tipo de contenido que realmente le funciona, o está desaprovechando el formato con mejor performance?

3. **Ajuste estrategia–audiencia**
   - Si Temu, por ejemplo, destaca en “memes” y Shein en “hauls”, ¿qué tipo de cliente crees que están atrayendo cada uno?  
   - ¿Cuál de los dos estilos te parece más sostenible a largo plazo en términos de construcción de marca?

4. **Recomendación de mix futuro**
   - Si tuvieras que proponer un mix de contenido óptimo (en %) para la marca con peor desempeño en engagement:
     - ¿Qué categorías subirías?  
     - ¿Qué categorías recortarías?  
     - ¿Qué experimentos A/B sugerirías (ej. más UGC vs más descuentos)?

5. **Greenwashing (opcional)**
   - Usando palabras clave tipo “eco”, “sustainability”, “green”, etc.:
     - ¿Qué porcentaje del contenido total habla de temas sostenibles?  
     - ¿Su engagement está por encima o por debajo del promedio general?  
     - ¿Dirías que la audiencia “compra” ese mensaje o lo ignora?


In [ ]:
import os
os.environ["OPENAI_API_KEY"] ="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"

In [ ]:
from openai import OpenAI

client = OpenAI()

In [ ]:
def call_llm(prompt):
  response = client.responses.create(
  model="gpt-4o-mini",
  input=prompt
)
  return response.output_text



In [ ]:
def clasify(text):
    prompt = (f"Clasifica el sentimiento del siguiente caption de Instagram como "
              f"'positivo', 'neutral' o 'negativo' siempre en minuscula. Solo responde con una palabra.\n\nCaption:\n{text}")
    time.sleep(2)  # Evitar rate limits
    return call_llm(prompt)

shein_sample = shein.sample(10, replace=True, random_state=42).copy()
shein_sample['content_type'] = shein_sample['caption'].apply(classify_content_type)
shein_sample[['caption', 'content_type']]


caption         content_type
102  @valentinavelasquezk nos tiene obsesionadas co...                  ugc
179  Miremos donde miremos, ¡todo son gangas! 🤩 No ...            descuento
92   Soñamos a lo grande, pero hoy vamos a hacerlo ...                 otro
14   ⚡ Outfit check! 👀🔥\n¿cuál es tu vibra hoy?@dan...                 haul
106  ¡El accesorio perfecto para este verano! 🍹\n\n...  producto individual
71   Cuida tu jardín y observa cómo florece con val...       sostenibilidad
188  Solo son cosas de chicas 😎 Desliza para conoce...                 haul
20   💅 Diferentes estilos de uñas postizas, ¡llenas...  producto individual
102  @valentinavelasquezk nos tiene obsesionadas co...                  ugc
121  De la ciudad al campo, con el set más cool 🌿✨🐄...                 otro

In [ ]:
def clasify(text):
    prompt = (f"Clasifica el sentimiento del siguiente caption de Instagram como "
              f"'positivo', 'neutral' o 'negativo' siempre en minuscula. Solo responde con una palabra.\n\nCaption:\n{text}")
    time.sleep(2)  # Evitar rate limits
    return call_llm(prompt)

temu_sample = temu.sample(10, replace=True, random_state=42).copy()
temu_sample['content_type'] = temu_sample['caption'].apply(classify_content_type)
temu_sample[['caption', 'content_type']]


caption         content_type
6   🌌👼🏻✨ ¿Quién dijo que volar es de ensueño?¡Solo...                 otro
19  ¡Libera tu creatividad con esta cinta decorati...  producto individual
14  ¡Deslumbra y brilla con nuestra exquisita cole...  producto individual
10  ❤️❤️Valen mucho la pena⏰ \n\nSolo para los nue...                 haul
7   ¿Harto de usar múltiples utensilios de limpiez...  producto individual
20  Eleva tu guardarropa con los tonos clásicos de...                 otro
6   🌌👼🏻✨ ¿Quién dijo que volar es de ensueño?¡Solo...                 otro
25  ¡Sube de nivel tu maquillaje diario con estas ...  producto individual
18  Añade un toque de elegancia y encanto a tu col...  producto individual
22  Ilumina tu espacio con esta hermosa lámpara fl...  producto individual

In [ ]:
shein_sample['content_type'].value_counts()

content_type
ugc                    2
otro                   2
producto individual    2
haul                   2
descuento              1
sostenibilidad         1
Name: count, dtype: int64

In [ ]:
temu_sample['content_type'].value_counts()

content_type
producto individual    6
otro                   3
haul                   1
Name: count, dtype: int64

In [ ]:
shein_sample.groupby('content_type')['engagement_rate'].mean()

content_type
descuento              0.041136
haul                   0.033186
otro                        NaN
producto individual         NaN
sostenibilidad              NaN
ugc                    0.050705
Name: engagement_rate, dtype: float64

In [ ]:
temu_sample.groupby('content_type')['engagement_rate'].mean()

content_type
haul                   0.048562
otro                        NaN
producto individual    0.034270
Name: engagement_rate, dtype: float64

## 6. Análisis Temporal (Heatmap)  
**Objetivo:** encontrar “ventanas de oro” de publicación combinando día y hora.

Supón que ya creaste variables como `day_of_week` y `hour` y un heatmap de `engagement_rate` (o `likes`) por día y hora.

### 6.1. Preguntas de negocio

1. **Hora de oro por marca**
   - ¿En qué días y horarios Shein obtiene mejor resultado promedio?
   ¿Y Temu?
   
   **R/ Shein: Publica sobre todo los días 4, 2 y 6 (jueves, martes y sábado). La franja horaria dominante es 14:00, muy por encima del resto (126 publicaciones).También aparece actividad secundaria a las 15:00. Hora de oro Shein: 14:00, especialmente entre semana (martes–jueves).**

   **Temu: Publica más en días 4 y 1 (jueves y lunes). Su franja dominante es 15:00, seguida por 14:00. Hora de oro Temu: 15:00 (y 14:00 como segunda mejor).**

    ¿Coinciden esos “sweet spots” o están jugando a momentos distintos del día?

    **R/ No coinciden exactamente: Shein rinde más a las 14:00, mientras Temu lo hace a las 15:00. Ambas marcas operan en momentos parecidos del día, pero no en la misma hora exacta.**

2. **Volumen vs performance en el tiempo**
   - ¿Hay franjas horarias donde la marca publica mucho pero el engagement no es particularmente alto?

   **R/ Shein publica MASIVAMENTE a las 14:00 (126 posts), pero eso no significa automáticamente alto engagement; el volumen exagerado suele bajar el ratio de eficiencia. A las 15:00, Shein también publica bastante, pero mucho menos que a las 14:00, lo cual puede significar menos saturación y mejor rendimiento relativo.**
  
     **Temu: La franja de 15:00 es dominante, pero no sabemos si el engagement acompaña ese volumen (normalmente no lo hace con tantos picos). A las 14:00 publica menos, lo cual puede dar una franja menos saturada.**


  Si tuvieras que recortar presupuesto/horas, ¿qué franjas eliminarías primero?

  **R/ Recortaría inversión y horas en las franjas donde la marca publica mucho pero sin evidencia de alto engagement, especialmente el exceso de contenido de Shein a las 14:00, que parece más saturación que eficiencia.**

3. **Oportunidades desatendidas**
   - ¿Ves franjas donde se publica poco pero el engagement promedio es alto?

   **Shein publica muy poco en horas como 1:00, 2:00, 3:00, 5:00, 6:00, 12:00, 16:00, pero no necesariamente tienen bajo engagement.**

   **Temu publica casi exclusivamente entre 12 y 15, dejando vacías muchas horas donde podría probar contenidos.**
  
  Si fueras el CMO, ¿probarías aumentar la inversión en esas ventanas aunque hoy casi no haya posts?

  **R/ Sí, hay franjas desatendidas (sobre todo en la mañana y tarde extendida) donde casi no publican. Como CMO probaría pilotos controlados en esas horas: poco riesgo, alta oportunidad de descubrir ventanas donde la competencia no está activa.**

4. **Decisión de campaña clave**
   - Si el fondo decide financiar una gran campaña puntual (por ejemplo, lanzamiento de colección o gran descuento):

   **Shein → Jueves (día 4) a las 14:00, Es su franja histórica de mayor actividad. Tiene mayor probabilidad de arrastre orgánico porque su audiencia ya está “entrenada” a consumir contenido a esa hora.**

   **Temu → Jueves (día 4) a las 15:00, Coincide con su mayor volumen y probablemente su mayor visibilidad natural.**


  ¿En qué día y hora la lanzarías para maximizar impacto orgánico?  
 Justifica usando el heatmap (no solo tu intuición).


 **R/ Los value_counts muestran que Shein maximiza visibilidad a las 14:00 y Temu a las 15:00, especialmente en los días 4 y 1 (jueves y lunes). Estas franjas son donde las marcas históricamente concentran más publicaciones, por lo tanto son sus ventanas naturales de engagement. Lanzar la campaña en esas horas maximiza el impacto orgánico porque coincide con los hábitos reales de consumo de cada audiencia.**


In [ ]:
shein["timestamp"] = pd.to_datetime(shein["timestamp"])
shein["day"] = shein["timestamp"].dt.day_of_week

In [ ]:
shein["day"].value_counts()

day
4    32
2    31
6    31
3    30
5    26
1    25
0    25
Name: count, dtype: int64

In [ ]:
shein["timestamp"] = pd.to_datetime(shein["timestamp"])
shein["hour"] = shein["timestamp"].dt.hour

In [ ]:
shein["hour"].value_counts()

hour
14    126
15     37
2       8
1       7
13      7
3       4
12      3
5       3
6       2
16      2
4       1
Name: count, dtype: int64

In [ ]:
temu["timestamp"] = pd.to_datetime(shein["timestamp"])
temu["day"] = shein["timestamp"].dt.day_of_week

In [ ]:
temu["day"].value_counts()

day
4    6
1    5
0    4
6    4
2    3
3    3
5    3
Name: count, dtype: int64

In [ ]:
temu["timestamp"] = pd.to_datetime(shein["timestamp"])
temu["hour"] = shein["timestamp"].dt.hour

In [ ]:
temu["hour"].value_counts()

hour
15    17
14     9
12     2
Name: count, dtype: int64

## 7. Preguntas de Negocio Finales (Entregable Ejecutivo)

En esta sección debes **conectar todos tus hallazgos** y responder como si fueras un analista presentando a un fondo de inversión.

### 7.1. Diferencia estratégica (posicionamiento)

Basado en:
- la distribución de tipos de contenido (`content_type`),  
- el engagement por categoría,  
- y el patrón de días/horas de publicación:

> **Describe en 5–7 líneas la estrategia de contenido de Shein vs Temu.**  
> ¿Qué hace cada una mejor? ¿En qué se diferencian claramente?

---

### 7.2. Eficiencia vs volumen

Usando tus resultados de `engagement_rate` y número de posts:

- ¿Quién está “ganando” la batalla por atención eficiente?  
- Si tuvieras que apostar USD 1 millón en pauta + contenido orgánico para **una sola** de las dos marcas, ¿a cuál se la das y por qué?

---

### 7.3. Hora de oro

Con base en tu análisis temporal (heatmap):

- ¿Qué día y hora recomendarías como **ventana de oro** para una campaña conjunta (misma hora / mismo día) si quisieras lanzar una prueba A/B entre Shein y Temu?  
- Explica por qué esa ventana maximiza probabilidad de éxito.

---

### 7.4. Greenwashing detector (opcional avanzado)

Filtrando contenido “eco” o “sostenible” (si lo analizaste):

- ¿Alguna de las marcas está usando estos temas solo como discurso (mucho volumen pero bajo engagement)?  
- ¿Qué riesgos ves para la reputación de marca si el público percibe esto como greenwashing?

---

### 7.5. Recomendación ejecutiva (1 párrafo)

Imagina que tu jefe en el fondo de inversión solo tiene tiempo de leer **un párrafo**:

> Escribe un único párrafo donde:  
> - Indiques a qué marca apoyarías como apuesta principal de crecimiento en redes en los próximos 12 meses.  
> - Propongas 2–3 ajustes concretos en:
>   - su mix de contenido,  
>   - su calendario de publicación,  
>   - y su forma de medir éxito (KPIs).


### Espacio para tus respuestas ejecutivas

Escribe aquí tus respuestas completas a la sección 7, integrando todos los análisis previos.

7.1. Diferencia estratégica (posicionamiento)

**Shein construye una estrategia más variada y aspiracional, combinando UGC, hauls, outfits y producto individual, lo que le permite conectar desde lifestyle hasta utilidad. Temu, en cambio, se enfoca en producto individual y contenido muy directo, más parecido a un catálogo optimizado para venta rápida. En engagement, Shein logra mejores tasas promedio, mientras Temu es más estable pero menos emocional. Además, Shein opera con mayor variabilidad de días y horas, mientras Temu concentra su actividad en pocas franjas muy tácticas. En resumen: Shein construye marca y comunidad, Temu empuja venta y practicidad.**

7.2. Eficiencia vs volumen

**Con un engagement_rate promedio de ~0.37 en Shein y ~0.036 en Temu, Shein es claramente más eficiente convirtiendo vistas en interacción real. Si tuviera que apostar USD 1 millón en pauta y contenido, invertiría en Shein, porque no solo publica más, sino que cada view rinde mucho mejor. Temu podría mejorar con optimización, pero hoy Shein ofrece un ROI más alto por pieza de contenido.**

7.3. Hora de oro

**La mejor ventana conjunta para lanzar una campaña A/B entre ambas marcas sería el jueves a las 14:00–15:00. Shein históricamente domina a las 14:00 y Temu en 15:00, por lo que esta franja permite activar simultáneamente a las dos audiencias en su momento de mayor consumo orgánico. Esta hora tiene mayor probabilidad de entrega algorítmica y maximiza visibilidad inicial, que es clave en un test A/B.**

7.5. Recomendación ejecutiva (1 párrafo)

**Recomiendo apostar por Shein como la marca principal para impulsar crecimiento en redes en los próximos 12 meses, dado su engagement significativamente superior y su mezcla de contenido más versátil y efectiva. Sugeriría tres ajustes clave: Reforzar los formatos que ya funcionan UGC, hauls y producto individual reduciendo el exceso de publicaciones redundantes; Optimizar su calendario concentrando esfuerzos en las ventanas de mayor rendimiento (especialmente jueves a las 14:00); Y migrar de métricas vanidosas a KPIs de eficiencia como engagement_rate por categoría y rendimiento por franja horaria. Estos cambios alinean volumen, creatividad y retorno, maximizando impacto orgánico y pagado.**